lib/tokenize.cpp

```diff
@@ -2537,7 +2537,7 @@ namespace {
                     tok1 = tok1->previous();
                 if (tok1->previous() && (tok1->strAt(-1) == ")" || tok->strAt(-1) == "}")) {
                     tok1 = tok1->linkAt(-1);
-                    if (Token::Match(tok1->previous(), "throw|noexcept (")) {
+                    if (tok1 && Token::Match(tok1->previous(), "throw|noexcept (")) {
```

Comment:
There is a check for unlinked `)|}` somewhere, so I wonder why it doesn't fire.

Reply:
`deleteInvalidTypedef()` messes up the links since it runs after `createLinks()`. Maybe`if (!tokOffset || tokOffset->isKeyword()) syntaxError(tok);` at tokenize.cpp:1257 is an option.

Reply:
The issues does not appear to be fully fixed by my change. Will look into that first and then look into your comment.

Reply:
This required a few more checks down the line.

Reply:
> Maybe`if (!tokOffset || tokOffset->isKeyword()) syntaxError(tok);` at tokenize.cpp:1257 is an option.
`isControlFlowKeyword()` seems to work.

Reply:
What line is that now? Or do you want to take over in a new PR?

Reply:
`tokenize.cpp:1257` is still correct,

Reply:
That is line 1255 in my tree...but that works.